.ghci

```diff
@@ -1,31 +1,28 @@
--- See docs/💡ProTip!.md
-:def! pretty \ _ -> return ":set -interactive-print Semantic.Util.Pretty.prettyShow"
-:def! no-pretty \_ -> return ":set -interactive-print System.IO.print"
-:def! r \_ -> return (unlines [":reload", ":pretty"])
-
--- See docs/💡ProTip!.md for documentation & examples.
-:{
-assignmentExample lang = case lang of
-  "Python" -> mk "py" "python"
-  "Go" -> mk "go" "go"
-  "Ruby" -> mk "rb" "ruby"
-  "JavaScript" -> mk "js" "typescript"
-  "TypeScript" -> mk "ts" "typescript"
-  "Haskell" -> mk "hs" "haskell"
-  "Markdown" -> mk "md" "markdown"
-  "JSON" -> mk "json" "json"
-  "Java" -> mk "java" "java"
-  "PHP" -> mk "php" "php"
-  _ -> mk "" ""
-  where mk fileExtension parser = putStrLn ("example: fmap (() <$) . runTask . parse " ++ parser ++ "Parser =<< Semantic.Util.blob \"example." ++ fileExtension ++ "\"") >> return ("import Parsing.Parser\nimport Semantic.Task\nimport Semantic.Util")
-:}
-:def! assignment assignmentExample
-
--- Enable breaking on errors for code written in the repl.
-:seti -fbreak-on-error
-
--- Continue loading after warnings when in the repl.
-:set -Wwarn
-
--- Use a cyan lambda as the prompt.
-:set prompt "\ESC[1;36m\STXλ \ESC[m\STX"
+-- Consider adding these to your ~/.ghc/ghci.conf file:
+
+-- -- Pretty-printing
+-- :set -package pretty-simple
+-- :def! pretty \ _ -> pure ":set -interactive-print Text.Pretty.Simple.pPrint"
+-- :def! no-pretty \ _ -> pure ":set -interactive-print System.IO.print"
+-- :def! r \_ -> pure ":reload\n:pretty"
+
+-- -- Turn on some language extensions you use a lot
+-- :seti -XFlexibleContexts -XOverloadedStrings -XTypeApplications
+
+-- -- Break on errors
+-- :seti -fbreak-on-error
+
+-- -- Automatically show the code around breakpoints
+-- :set stop :list
+
+-- -- Use a cyan lambda as the prompt
+-- :set prompt "\ESC[1;36m\STXλ \ESC[m\STX"
+
+-- -- Better errors
+-- :set -ferror-spans -freverse-errors -fprint-expanded-synonyms
+
+-- -- Path-local ghci history
+-- :set -flocal-ghci-history
+
+-- -- Better typed holes
+-- :set -funclutter-valid-hole-fits -fabstract-refinement-hole-fits -frefinement-level-hole-fits=2
```

Comment:
We should be defining these in our personal .ghci files because if we define them here then it can get super confusing as to what’s being defined where and what’s overriding what. Pretty-printing especially belongs out of here, so that we don’t end up loading multiple versions of the pretty-printing package and throwing errors on every reload. (Ask me how I know about that one.)

Reply:
I remember that one. That one bit me during a demo. 😅

Reply:
Since this file is now one big comment, can we uncomment the relevant `:set` calls and move this into a `.ghci.sample` file, so we can `cp .ghci.default ~/.ghc/.ghci`?

Reply:
That’s a superb idea, thank you! 👍